codegen/casing.go

```diff
@@ -71,6 +71,17 @@ var CommonInitialisms = map[string]bool{
 }
 var camelingRegex = regexp.MustCompile("[0-9A-Za-z]+")
 
+// CorrectMethodNaming correct the naming for initialisms
+func CorrectMethodNaming(key string) string {
```

Comment:
This function works on a string whose first letter is in upper case and rest are in lower case, imagine a method name that is `myService::uuidTest`, it won't work anymore. The target `key` is a string whose upper case is in the common intialisms.  There is already the `ensureGolangAncronymCasing` method that you can leverage to come up a generic function that works on an arbitrary string. It's also not limited to method naming, so the function name should be generic.

Reply:
+1 on more case check required but scope could go wild

for example `API` check too general might mess up with name like
`MyServiceApicultures` -> `MyServiceAPIcultures` ?

if this can be extract to easily extended rules instead of hardcode check it would be better, we'll just be patching the cases we see

on the other side, this was the concern I had whether we should fix this or just bypass lint